pkg/batch/scan.go

```diff
@@ -22,18 +23,55 @@ type Options struct {
 type Batch struct {
 	Data     net.Buffers
 	Location Location
+
+	// backup hold the same data as Data. It is used to rewind if something goes wrong
+	// Because it copies the slice, the memory is not duplicated
+	// Because we only read data, the underlaying memory is not modified neither
+	backup net.Buffers
+}
+
+func NewBatch(data net.Buffers, location Location) Batch {
+	b := Batch{
+		Data:     data,
+		Location: location,
+	}
+	b.snapshot()
+	return b
+}
+
+func (b *Batch) snapshot() {
+	b.backup = net.Buffers{}
+	b.backup = append(b.backup, b.Data...)
+}
+
+// Makes data available again to read
+func (b *Batch) Rewind() {
+	b.Data = net.Buffers{}
+	b.Data = append(b.Data, b.backup...)
 }
 
 // Location positions a batch within the original data
 type Location struct {
+	// StartRow represents the index of the row where the batch starts.
+	// First row of the file is row 0
+	// The header counts as a line
 	StartRow int64
-	Length   int
+	// RowCount is the number of rows in the batch
+	RowCount int
+	// ByteOffset is the byte position on the original file.
+	// It can be used with ReadAt to process the same batch again.
+	ByteOffset int
+	// ByteLen represents the number of bytes for the batch.
+	// It can be used to know how big the batch is and read it accordingly
+	ByteLen int
 }
 
-func NewLocation(rowsRead int64, bufferedRows int, skip int) Location {
+func NewLocation(rowsRead int64, bufferedRows int, skip int, byteOffset int, byteLen int) Location {
```

Comment:
Usually I'm against defining the type only once, it may be sorter to type but takes longer to read.

rowsRead was an int64, provably not for a good reason. 
Maybe we could make it int64, but I don't think it makes any difference and provably should be addressed on a different PR